warehouse/query-core/src/main/java/datawave/query/index/lookup/IndexInfo.java

```diff
@@ -389,14 +392,31 @@ protected IndexInfo intersect(long maxPossibilities, Iterable<IndexMatch> matchI
         }
         
         JexlNode newNode;
-        if (matches.size() > 1) {
+        if (matches.isEmpty()) {
+            JexlNodeSet nodes = new JexlNodeSet();
+            if (matchNode != null) {
+                nodes.add(matchNode);
+            }
+            if (!otherInfiniteNodes.isEmpty()) {
+                nodes.addAll(otherInfiniteNodes);
+            }
+            if (!delayedNodes.isEmpty()) {
+                nodes.addAll(delayedNodes);
```

Comment:
interesting that we were missing these before.  I wonder what misses that was resulting in?

Reply:
Persisting delayed terms in uid matches was handled elsewhere...I think.